src/privileged/blurts/FirefoxMonitor.jsm

```diff
@@ -0,0 +1,716 @@
+  }
+
+  panel.addEventListener("popuphidden", () => {
+    showPanel(browser, ui, notificationId, telemetryId);
+  }, { once: true });
+}
+
+function showPanel(browser, ui, notificationId, telemetryId) {
+  let doc = browser.ownerDocument;
+  let panel = doc.defaultView.PopupNotifications.panel;
+
+  let populatePanel = (event) => {
+    if (event !== "shown") {
+      return;
+    }
+    let n = doc.getElementById(notificationId + "-notification");
+    let body = doc.getAnonymousElementByAttribute(n, "class", "popup-notification-body");
+    let box = body.querySelector(".blurtsbox");
+    if (box) {
+      box.remove();
+    }
+    box = ui.box;
+    box.setAttribute("class", "blurtsbox");
+    box.setAttribute("style", "font-size: 110%");
+    body.appendChild(box);
+    let icon = doc.getAnonymousElementByAttribute(n, "class", "popup-notification-icon");
+    if (icon) {
+      icon.remove();
+    }
+  };
+
+  doc.defaultView.PopupNotifications.show(
+    browser, notificationId, "",
+    null, ui.primaryAction, ui.secondaryActions, {persistent: true, hideClose: true, eventCallback: populatePanel});
+  FirefoxMonitor.notifyEventListeners(`${telemetryId}_shown`);
+}
+
+const XUL_NS = "http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul";
+let UIFactory = [
+  function(browser, doc, host, site) {
+    return {
+      get box() {
+        let box = doc.createElementNS(XUL_NS, "vbox");
+        let elt = doc.createElementNS(XUL_NS, "hbox");
+        elt.setAttribute("style", "margin-bottom: 1rem;");
+        elt.setAttribute("pack", "center");
+        let imgElt = doc.createElementNS(XUL_NS, "image");
+        imgElt.setAttribute("src", imageDataURIs.iconLaptopRisk);
+        imgElt.setAttribute("style", "width: 148px;");
+        elt.appendChild(imgElt);
+        box.appendChild(elt);
+        elt = doc.createElementNS(XUL_NS, "description");
+        elt.setAttribute("anonid", "maindesc");
+        elt.appendChild(doc.createTextNode("Have an account on this website?\nIt may be at risk."));
+        elt.setAttribute("style", "font-size: 150%; text-align: center; white-space: pre; margin-bottom: 1rem;");
+        box.appendChild(elt);
+        elt = doc.createElementNS(XUL_NS, "description");
+        elt.appendChild(doc.createTextNode(site.Title + " was reported to Firefox Monitor, a service that collects information about data breaches and other ways hackers can steal your information.\n\nFind out if your account is at risk and learn how to protect yourself at the Firefox Monitor website."));
+        elt.setAttribute("style", "text-align: center; white-space: pre-wrap;");
+        box.appendChild(elt);
+        return box;
+      },
+      primaryAction: {
+        label: "Go to Firefox Monitor",
+        accessKey: "f",
+        callback: () => {
+          FirefoxMonitor.notifyEventListeners(`variant_1_submit`);
+          doc.defaultView.openUILinkIn("http://fx-breach-alerts.herokuapp.com/?breach=" + site.Title, "tab");
```

Comment:
Sadly, i'm not 100% certain that `site.Title` will get us the correct results.

Consider these two http://bell.ca breaches (from the src/breaches.json file in this PR):

```
{"Domain":"bell.ca","Title":"Bell (2014 breach)","BreachDate":"2014-02-01","PwnCount":20902,"DataClasses":["Credit cards","Genders","Passwords","Usernames"],"LogoFilename":"Bell.svg"},
{"Domain":"bell.ca","Title":"Bell (2017 breach)","BreachDate":"2017-05-15","PwnCount":2231256,"DataClasses":["Email addresses","Geographic locations","IP addresses","Job titles","Names","Passwords","Phone numbers","Spoken languages","Survey results","Usernames"],"LogoFilename":"Bell2017.svg"},
```

And here are the 2 server endpoints we need to redirect to:
- https://blurts-server.stage.mozaws.net/?breach=Bell (aka: "Bell Breach")
- https://blurts-server.stage.mozaws.net/?breach=Bell2017 (aka: "Bell2017 Breach")

If you solely use `site.Title`, wouldn't it try redirecting to:
blurts-server.stage.mozaws.net/?breach=Bell+(2014+breach)

But I'm generally confused, because this is confusing and I don't see any great unique identifiers... apart from using site.LogoFilename (minus the file extension, which I _think_ is what the server half is somewhat expecting).

Reply:
Actually, I guess I filed #20 which sort of covers this. Or basically, if a site has multiple breaches, we'd only ever tell them about one of them.

Reply:
Thoughts on this : why not send the domain name (or whatever is stored in "Domain") to the server and create 'domain' as a column in the server database? For now, we can query by the "domain" value and if there are multiple breaches, we can pull the first one and either 
1.) opt to just show only the first breach (for now) in acceptance of the fact that we very probably don't have time to have an "all breaches associated with this company" page mocked up and ready to go  
OR  2.) show the first breach on the banner like usual but with a message saying something to the effect of "This website is associated with additional breaches. Click HERE to see all of them." Clearly, the UI guys and gals would have to jazz that up but you get the idea.